File: onezero.py
class Queue:
    def __init__(self):
        self.items = []

    def isEmpty(self):
        return self.items == []

    def enqueue(self, item):
        self.items.insert(0,item)

    def dequeue(self):
        return self.items.pop()

    def size(self):
        return len(self.items)

num = []
num.append(1)
q = Queue()
cur_num = 1
while cur_num < 20000:
    # print cur_num
    cur_num_0 = int(str(cur_num)+'0')
    cur_num_1 = int(str(cur_num)+'1')
    q.enqueue(cur_num_0)
    q.enqueue(cur_num_1)
    cur_num = q.dequeue()


t = input()
while t:
    sankhya = input()
    for value in num:
        cur_num = value
        if(value >= sankhya):
            break
    q = Queue()
    while cur_num % sankhya != 0:
        cur_num_0 = int(str(cur_num)+'0')
        cur_num_1 = int(str(cur_num)+'1')
        q.enqueue(cur_num_0)
        q.enqueue(cur_num_1)
        cur_num = q.dequeue()
    print cur_num
    t = t-1

File: onezero.cpp
#include <iostream>
#include <queue>
#include <stack>
#include <algorithm>
using namespace std;

int main(){
	ios::sync_with_stdio(false);
	int t;
	cin>>t;
	while(t--){
		int num;
		cin>>num;
		int cur_num = 1;
		int parent[num], value[num];
		fill_n(parent, num, -1);
		parent[1] = 0;
		value[1] = 1;
		queue<int> q;
		while(cur_num % num != 0){
			int cur_num_0 = (cur_num*10) % num;
			if(parent[cur_num_0] == -1){
				parent[cur_num_0] = cur_num;
				value[cur_num_0] = 0;
				q.push(cur_num_0);
				// cout<<"parent of "<<cur_num_0<<" is :"<<cur_num<<endl;
			}
			int cur_num_1 = (cur_num_0+1) % num;
			if(parent[cur_num_1] == -1){
				parent[cur_num_1] = cur_num;
				value[cur_num_1] = 1;
				q.push(cur_num_1);
				// cout<<"parent of "<<cur_num_1<<" is :"<<cur_num<<endl;
			}
			cur_num = q.front();
			// cout<<"popping: "<<cur_num<<endl;
			q.pop();
		}
		stack<int> s;
		while(parent[cur_num] != 0){
			s.push(value[cur_num]);
			cur_num = parent[cur_num];
		}
		cout<<"1";
		while(!s.empty()){
			cout<<s.top();
			s.pop();
		}
		cout<<endl;
	}
	return 0;
}